clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -175,7 +175,7 @@ public Selector(long connectionMaxIdleMS, Metrics metrics, Time time, String met
      */
     @Override
     public void connect(String id, InetSocketAddress address, int sendBufferSize, int receiveBufferSize) throws IOException {
-        if (this.channels.containsKey(id))
+        if (this.channels.containsKey(id) || this.closingChannels.containsKey(id))
```

Comment:
Maybe we want to have a different message for the second case? It may give more clues if we end up running into it.